06x-new-kafka-roller.md

```diff
@@ -0,0 +1,195 @@
+- `Restart` - Delete a pod via Kubernetes API and then `Wait and Observe`. This action is followed by preferred leader election trigger via Kafka Admin API for a broker node if it is not leading its preferred replicas yet.
+- `Reconfigure` - Apply configuration updates via Kafka Admin API and then `Wait and Observe`.
+- `No action` - This means we reached the desired state after taking one of the above actions or no action is needed.
+
+
+### Observations -> States Map
+ | KubeAPI | KafkaAgent | States |
+ | :--------------- | :--------------- | :---------------
+ | - | - | UNKNOWN
+ | Pod is not Running | - | NOT_RUNNING 
+ | Pod is Running but lacking Ready status | Broker state != 2 | NOT_READY 
+ | Pod is Running but lacking Ready stats | Broker state == 2 | RECOVERING 
+ | Pod is Running and Ready | - | READY
+
+### States -> Actions Map
+ | States | Actions 
+ | :--------------- | :--------------- 
+ | UNKNOWN | `Observe`
+ | NOT_RUNNING | `Restart` OR `Wait and observe`
+ | RECOVERING | `Wait and observe`
+ | NOT_READY | `Restart` OR `Wait and observe`
+ | READY | `Restart` OR `Reconfigure` OR `No action`
+
+Some states map to multiple possible actions, but only one of them is taken based on the other conditions.
+
+`UNKNOWN` nodes will be observed. This is the initial/default state before observation. 
+
+`NOT_RUNNING` nodes will restarted only if they have `POD_HAS_OLD_REVISION`. This is because, if the node is not running at all, then restarting it likely won't make any difference unless the node is out of date. For example, if a pod is in pending state due to misconfigured affinity rule, there is no point restarting this pod again or restarting other pods, because that would leave them in pending state as well. If the user then fixed the misconfigured affinity rule, then we should detect that the pod has an old revision, therefore should restart it so that pod is scheduled correctly and runs.
+
+`RECOVERING` nodes will be waited and observed only. A Kafka node can take a long time to become ready while performing log recovery and it's not easy to determine how long it might take. Therefore, it's important to avoid restarting the node during this process, as doing so would restart the entire log recovery, potentially causing the node to enter a loop of continuous restarts without becoming ready. Moreover, while a node is in recovery, no other node should be restarted, as this could impact cluster availability and affect the client.
+
+`NOT_READY` nodes will be restarted if they have a restart reason and have not been restarted yet. If it is not ready after being restarted already, we don't want to restart any other nodes to avoid taking down more nodes. 
+
+`READY` nodes will be restarted if they have a restart reason. If they don't have a restart reason but need to be reconfigured, they will be reconfigured. If no reconfiguration is needed, then no action will be taken on these nodes. 
```

Comment:
If we want to stop a roll, would the process be to apply `strimzi.io/pause-reconciliation="true"`? Would that put each node in READY or something else?